### PR TITLE
feat(svelte): Improve small screeen behavior

### PR DESCRIPTION
**Pull request by fkling**
Closes srch-730

This is an initial attempt to improve the web app for small screens. This commit makes the following changes for small screens:

- Search home page:
  - No more search input overflow
  - History button is rendered on same line as other action buttons (saves some vertical space)
- Search results page:
  - Search input is rendered in the page, not in the header (more space)
  - Filters sidebar is hidden by default and can be shown via a `Filters` button.
  - The filters sidebar opens fullscreen and has a close button
  - The progrss button is smaller due to showing less information
- Repository pages:
  - File sidebar is hidden by default. It can be shown via a new button that is visible in the file headers
  - The file sidebar opens fullscreen and has a close button
  - NOTE: Selecting a file currently does not close the file sidebar, navigation happens in the background
  - Cody sidebar opens fullscreen and has the same close button
- General:
  - Fuzzy finder opens fullscreen and has a larger close button
  - Tabs don't show keyboard shortcuts

I tried to stick to CSS as much as possible but for some things to work I had to change component structures or rendered elements conditionally. Specifically when a component was already using `$isViewportMobile` I usually just rendered elements conditionally.

I extended the `Panel` component to have a special 'mobile' mode, since I realized I was doing similar changes to the filters, file tree and cody sidebar.
The cody sidebar is a bit of a special case though because it's not even rendered by default. So there are some additional steps required to sync the open state.

Screenshots (iPhone SE, which is one of the smaller phones I guess)

| Situation | Before | After |
|--------|--------|--------|
| Search home |  |  |
| Search results |  |  |
| Filters |   |  |
| Repo |  |  |
| File sidebar |  |  | 
| Cody sidebar |  |  | 
| Repo search |  |  | 
| Fuzzy finder |  |  | 
| Rev picker |  |  |

Me rambling about the changes:

https://github.com/user-attachments/assets/670f7764-0ef2-4f1b-bc33-89a86d4b2274

Note that the commits and commit pages already seem to look fine. The branches, tags and contributors pages are a bit broken due to use of tables and fixed columns. I can look at those separately.

## Test plan

Manual testing

### CODE REVIEW

File: client/web-sveltekit/src/lib/dom.ts [hunk 1]
```diff
@@ -161,7 +162,18 @@ export const popover: Action<HTMLElement, { reference: Element; options: Popover
...
+            flip(options.flip),
+            size({
+                apply({ availableWidth, availableHeight }) {
+                    Object.assign(popover.style, {
+                        maxWidth: `${availableWidth}px`,
+                        maxHeight: `${availableHeight}px`,
+                    })
+                },
+            })
+        )
```
Comment (fkling) [inline client/web-sveltekit/src/lib/dom.ts:176]: I added this to resize popovers vertically.

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/FileTree.svelte [hunk 2]
```diff
@@ -122,10 +122,18 @@
...
+                    placement="right-start"
+                    showOnHover
+                    offset={{ mainAxis: 8, crossAxis: -32 }}
+                >
                     <a
                         href={replaceRevisionInURL(entry.canonicalURL, revision)}
-                        on:click|preventDefault={() => {}}
+                        on:click|preventDefault={() => {
+                            toggle(false)
+                        }}
```
Comment (fkling) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/FileTree.svelte:136]: Instead of having `Popover` hide on click when `showOnHover` is on, we do it manually here.

File: client/web-sveltekit/src/lib/Popover.svelte [hunk 3]
```diff
@@ -94,15 +95,13 @@
         trigger.addEventListener('mouseenter', handleMouseEnterTrigger)
         trigger.addEventListener('mouseleave', handleMouseLeaveTrigger)
         trigger.addEventListener('mousemove', handleMouseMoveTrigger)
-        trigger.addEventListener('click', close)
```
Comment (fkling) [inline client/web-sveltekit/src/lib/Popover.svelte:97]: This interferes with having popovers be conditionally by triggered by click or hover. I think callsites should be responsible for hiding the popover as necessary.

**Review by camdencheek — APPROVED** [review]
This is so much more usable -- thank you!

File: client/web-sveltekit/src/lib/TabsHeader.svelte [hunk 4]
```diff
@@ -130,4 +134,12 @@
...
         }
     }
+
+    span.shortcut {
+        display: contents;
+
+        @media (--mobile) {
+            display: none;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (fkling) [reply client/web-sveltekit/src/lib/TabsHeader.svelte:144]: Probably. I can move that into the keyboard shortcut component instead.

File: client/web-sveltekit/src/lib/dom.ts [hunk 5]
```diff
@@ -230,6 +230,10 @@ export const restrictToViewport: Action<HTMLElement, { offset?: number }> = (nod
 
     window.addEventListener('resize', setMaxHeight)
 
+    // Also update when the content changes
+    const observer = new MutationObserver(setMaxHeight)
+    observer.observe(node, { childList: true, subtree: true })
```
[the remark this answers is not in the document]
Comment (camdencheek) [reply client/web-sveltekit/src/lib/dom.ts:247]: Oooh, gotcha. Makes sense, thanks

File: client/web-sveltekit/src/lib/navigation/GlobalHeader.svelte [hunk 6]
```diff
@@ -122,10 +122,10 @@
 
     <div class="global-portal" bind:this={$extensionElement} />
 
-    <Popover let:registerTrigger showOnHover hoverDelay={100} hoverCloseDelay={50}>
-        <span class="web-next-badge" use:registerTrigger>
+    <Popover let:registerTrigger let:toggle showOnHover={!$isViewportMobile} hoverDelay={100} hoverCloseDelay={50}>
+        <button class="web-next-badge" use:registerTrigger on:click={() => toggle()}>
             <Badge variant="warning">Experimental</Badge>
-        </span>
+        </button>
```
Comment (fkling) [inline client/web-sveltekit/src/lib/navigation/GlobalHeader.svelte:128]: this converts the opt out badge into a button that works on click on mobile (since hover is disabled).

File: client/web-sveltekit/src/lib/wildcard/resizable-panel/Panel.svelte
Comment (fkling) [inline client/web-sveltekit/src/lib/wildcard/resizable-panel/Panel.svelte:1]: The changes here are quite "rudimentary". I don't know how they affect the panel system though it shouldn't matter much on mobile.
Still it would be nice to think about properly supporting this use case and not just tack it on.

**Comment by fkling** [comment]
Made some additional updates to make the revision picker work. For this I added an additional option to the default popover config that allows the popover logic to move the popover to the cross axis.
It doesn't open fullscreen which deviates from the behavior of the other panels (fullscreen with close button), but that might be OK since it's not a panel and not a dialog either.

| Before | After |
|--------|-------|
|  |  |

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/RepositoryRevPicker.svelte [hunk 7]
```diff
@@ -231,13 +236,23 @@
...
+                grid-template-columns: 1fr;
+            }
         }
 
         :global([data-picker-suggestions-list-item]) {
             display: grid;
-            grid-column: 1 / 4;
+            grid-column: 1/4;
             grid-template-columns: subgrid;
+            grid-template-areas: 'title author timestamp';
```
Comment (fkling) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/RepositoryRevPicker.svelte:249]: Not sure if that's the best way to go about. It requires that we set `grid-area:` in the other component. But maybe that's OK since they are related.

File: client/web-sveltekit/src/lib/Popover.svelte [hunk 8]
```diff
@@ -178,7 +182,7 @@
         border: 1px solid var(--dropdown-border-color);
         border-radius: var(--popover-border-radius);
         // Ensure child elements do not overflow the border radius
-        overflow: hidden;
+        overflow-y: scroll;
```
Comment (vovakulikov) [inline client/web-sveltekit/src/lib/Popover.svelte:185]: Is it impossible to have overflow by X axes? What is the reason for this change

File: client/web-sveltekit/src/lib/search/input/SearchInput.svelte [hunk 9]
```diff
@@ -110,13 +108,24 @@
...
+    export const enum Style {
+        /**
+         * Reduced paddings and margins.
+         */
+        Compact = 1 << 0,
+        /**
+         * No border around the input.
+         */
+        NoBorder = 1 << 1,
+    }
```
[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/web-sveltekit/src/lib/search/input/SearchInput.svelte:121]: Every time when someone uses a bitmask something good wakes up in my soul, thanks

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/RepositoryRevPicker.svelte [hunk 10]
```diff
@@ -231,13 +236,23 @@
...
             gap: 1rem;
+
+            @media (--mobile) {
+                gap: 0.5rem;
+                grid-template-areas: 'title timestamp' 'author author';
```
Comment (vovakulikov) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/RepositoryRevPicker.svelte:254]: You split this into two strings, I'm just curious: Is this some new grid syntax? I thought that there should be no difference.

**Review by vovakulikov — APPROVED** [review]

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/RepositoryRevPicker.svelte (hunk 10 above)
Comment (camdencheek) [reply client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/RepositoryRevPicker.svelte:254]: Each string is a space separated row, so this means:
```
```

File: client/web-sveltekit/src/lib/Popover.svelte (hunk 8 above)
Comment (fkling) [reply client/web-sveltekit/src/lib/Popover.svelte:185]: With the new `size` middleware the actual size of the popover might be different from what the popover content expects or there might simply not enough space.
E.g the content of the search results progress button was too large to completely show vertically. In this case the popover should be scrollable.

You are right that this might also be the case horizontally, although in this case I'd expect the popover content to simply shrink/wrap.

**Comment by fkling** [comment]
I changed the keyboard shortcuts component to not render on mobile and update the history panel and diff view to look a bit better:

| Before | After |
|--------|--------|
|  |  |
